website/docs/tutorials/06-deploying/02-migrations/declare.md

```diff
@@ -14,7 +14,16 @@ class DeclaredContract:
 Sends a declare transaction.
 
 - `contract_path_str` — A path to the **compiled** contract or a [contract name](../../compiling#contract-name). If you provide a contract name, Protostar will compile the contract to the migration output directory.
-- `config` — A keyword only argument that allows passing [network configuration](../03-network-config.md).
+- `config` — A keyword only argument that allows configuring transaction and network parameters.
+
+## `DeclareCheatcodeNetworkConfig`
+```python
+class DeclareCheatcodeNetworkConfig(CheatcodeNetworkConfig):
+    max_fee: NotRequired[Wei]
+```
+- `max_fee` — The maximum fee that the sender is willing to pay for the transaction.
```

Comment:
I wonder if we should allow for silently interpreting lack of max_fee as `auto_estimate`. I know that requiring the user to pass `auto_estimate` explicitly can make the UX slightly worse, but user should be aware of the mechanism he is using in this case, since it can incur large costs depending on the gas price. WDYT?

Reply:
It could also be made interactive, like: 
`This tx will cost you X wei, do you want to proceed?` (if the terminal is interactive) 
Else we should probably require one of `max_fee` or `auto_estimate`

Reply:
I think we should avoid interactive terminals if possible.

Reply:
<Internal discussion> -> `max_fee="auto"`

Reply:
Added `max_fee="auto"` support.